xknx/devices/climate.py

```diff
@@ -13,22 +13,65 @@
 from .remote_value_switch import RemoteValueSwitch
 from .remote_value_temp import RemoteValueTemp
 
+DEFAULT_SETPOINT_SHIFT_STEP = 0.5
+DEFAULT_SETPOINT_SHIFT_MAX = 6
+DEFAULT_SETPOINT_SHIFT_MIN = -6
+
+
+class SetpointShiftValue(RemoteValue1Count):
+    """Class for managing setpoint_shift values."""
+
+    def __init__(self,
+                 xknx,
+                 group_address=None,
+                 group_address_state=None,
+                 device_name=None,
+                 after_update_cb=None,
+                 setpoint_shift_step=DEFAULT_SETPOINT_SHIFT_STEP,
+                 min_kelvin=DEFAULT_SETPOINT_SHIFT_MIN,
+                 max_kelvin=DEFAULT_SETPOINT_SHIFT_MAX):
+        """Initialize SepointShift class."""
+        # pylint: disable=too-many-arguments
+        super().__init__(xknx,
+                         group_address,
+                         group_address_state,
+                         device_name,
+                         after_update_cb)
+
+        self.setpoint_shift_step = setpoint_shift_step
+        self.min_kelvin = min_kelvin
+        self.max_kelvin = max_kelvin
+
+    @property
+    def value(self):
+        """Return current value in Kelvin."""
+        if super().value is not None:
+            return super().value * self.setpoint_shift_step
+        return None
+
+    async def set(self, value):
+        """Set new value from Kelvin."""
+        if value > self.max_kelvin:
+            raise DeviceIllegalValue("setpoint_shift_max exceeded", value)
+            # value = self.max_kelvin
```

Comment:
we could log a warning and move on with the maximum value.

Reply:
I think this makes sense.